packages/filesystem/src/browser/filesystem-preferences.ts

```diff
@@ -80,24 +80,30 @@ These have precedence over the default associations of the languages installed.'
             markdownDescription: 'Controls the max file size in MB which is possible to open.'
         },
         'files.trimTrailingWhitespace': {
-            'type': 'boolean',
-            'default': false,
-            'description': 'When enabled, will trim trailing whitespace when saving a file.',
-            'scope': 'language-overridable'
+            type: 'boolean',
+            default: false,
+            description: 'When enabled, will trim trailing whitespace when saving a file.',
+            scope: 'language-overridable'
+        },
+        'files.maxConcurrentUploads': {
+            type: 'integer',
+            default: 1,
+            description: 'Maximum number of concurrent files to upload when uploading multiple files.',
```

Comment:
Setting this preference to something <1 leads to this unexpected behavior when trying to upload a file:

1. The 'Uploading Files' notification pops up for a few frames.
2. The notification disappears and an error is logged, since the `AsyncQueue` will throw.

I think we should handle this more gracefully. Does Theia support the notion of using only positive integers as preferences? Alternatively check for a value <1 in the `FileUploadService` and either 1. Behave if it was simply set to `1` or 2. Display a warning and ignore the dd event.

Reply:
Ah yes, forgot to gracefully handle that case.

> Does Theia support the notion of using only positive integers as preferences?

Not AFAIK. I made it so it will fall back on its feet no matter what, using the default if invalid.